src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -40,21 +49,25 @@ public static IDataProtectionProvider Create(DirectoryInfo keyDirectory)
             {
                 throw new ArgumentNullException(nameof(keyDirectory));
             }
-
             if (setupAction == null)
             {
                 throw new ArgumentNullException(nameof(setupAction));
             }
 
+            return CreateProvider(keyDirectory, setupAction);
+        }
+
+        private static IDataProtectionProvider CreateProvider(
+            DirectoryInfo keyDirectory,
+            Action<IDataProtectionBuilder> setupAction)
+        {
             // build the service collection
             var serviceCollection = new ServiceCollection();
             var builder = serviceCollection.AddDataProtection();
+
             builder.PersistKeysToFileSystem(keyDirectory);
```

Comment:
@PinpointTownes, this is a private method. When `keyDirectory` is null it will be caught here https://github.com/aspnet/DataProtection/pull/138/files#diff-ea6431c57866be038e1a9a1d98b14916R48